### PR TITLE
No longer showing "Scramble Set A" if there is only one scramble set/group

### PR DESCRIPTION
**Pull request by Shadowjockey**
closes Issue #433

### CODE REVIEW

File: webscrambles/src/main/kotlin/org/worldcubeassociation/tnoodle/server/webscrambles/wcif/model/ActivityCode.kt [hunk 1]
```diff
@@ -40,13 +40,13 @@ data class ActivityCode(val activityCodeString: String) : EventIdProvider {
...
 
-    fun compileTitleString(includeEvent: Boolean = true): String {
+    fun compileTitleString(includeEvent: Boolean = true, includeGroupID: Boolean = false): String {
         val parts = structureParts.map { (k, v) ->
             val translatePrefix = PREFIX_TRANSLATIONS[k]
             val translateValue = v.takeUnless { k == WCIF_PREFIX_GROUP }
                 ?: v.toIntOrNull()?.toColumnIndexString()
 
-            "$translatePrefix $translateValue"
+            if (!includeGroupID && k == WCIF_PREFIX_GROUP) "" else "$translatePrefix $translateValue"
```
Comment (suushiemaniac) [inline webscrambles/src/main/kotlin/org/worldcubeassociation/tnoodle/server/webscrambles/wcif/model/ActivityCode.kt:49]: Suggestion concerning Kotlin stdlib convenience:
`"myString".takeUnless { someBoolean }.orEmpty()`

**Comment by Shadowjockey** [comment]
@suushiemaniac should be fixed now.

File: webscrambles/src/main/kotlin/org/worldcubeassociation/tnoodle/server/webscrambles/wcif/model/ScrambleSet.kt [hunk 2]
```diff
@@ -5,7 +5,8 @@ import org.worldcubeassociation.tnoodle.server.webscrambles.wcif.model.extension
 import org.worldcubeassociation.tnoodle.server.webscrambles.wcif.provider.IndexingIdProvider
 
 @Serializable
-data class ScrambleSet(override val id: Int, val scrambles: List<@Serializable(with = Scramble.Companion::class) Scramble>, val extraScrambles: List<@Serializable(with = Scramble.Companion::class) Scramble>, override val extensions: List<Extension> = emptyList()) : ExtensionProvider(), IndexingIdProvider {
+data class ScrambleSet(override val id: Int, val scrambles: List<@Serializable(with = Scramble.Companion::class) Scramble>, val extraScrambles: List<@Serializable(with = Scramble.Companion::class) Scramble>, override val extensions: List<Extension> = emptyList()
```
Comment (suushiemaniac) [inline webscrambles/src/main/kotlin/org/worldcubeassociation/tnoodle/server/webscrambles/wcif/model/ScrambleSet.kt:8]: If you line-break this thing at all, please consider line-breaking per field:

```
data class ScrambleSet(
    val foo: String,
    val bar: Boolean,
    val baz: List<Int>
) : InheritedClass()
```

File: webscrambles/src/main/kotlin/org/worldcubeassociation/tnoodle/server/webscrambles/wcif/model/ScrambleSet.kt [hunk 3]
```diff
@@ -5,7 +5,8 @@ import org.worldcubeassociation.tnoodle.server.webscrambles.wcif.model.extension
...
+) : ExtensionProvider(), IndexingIdProvider {
```
Comment (suushiemaniac) [inline webscrambles/src/main/kotlin/org/worldcubeassociation/tnoodle/server/webscrambles/wcif/model/ScrambleSet.kt:9]: Also, please try to figure out why IntelliJ doesn't do the automated line breaking like outlined above (or alternatively discard this change from this particular commit, adding only one weeny line break seems silly given the length of the line)

File: webscrambles/src/main/kotlin/org/worldcubeassociation/tnoodle/server/webscrambles/wcif/model/ScrambleSet.kt (hunk 2 above)
Comment (Shadowjockey) [reply webscrambles/src/main/kotlin/org/worldcubeassociation/tnoodle/server/webscrambles/wcif/model/ScrambleSet.kt:8]: Oh yeah I failed to restore the original indentation

File: webscrambles/src/main/kotlin/org/worldcubeassociation/tnoodle/server/webscrambles/wcif/model/ScrambleSet.kt (hunk 3 above)
Comment (Shadowjockey) [reply webscrambles/src/main/kotlin/org/worldcubeassociation/tnoodle/server/webscrambles/wcif/model/ScrambleSet.kt:9]: It seems there's an option for that in editorconfig https://github.com/editorconfig/editorconfig/wiki/EditorConfig-Properties but it's under "limited support"

**Comment by suushiemaniac** [comment]
I had to make some changes in order to accomodate for broken schedules. Please `rebase` on top of master at your earliest convenience (we try to avoid `merge` in this project)

**Comment by suushiemaniac** [comment]
Nevermind, figured out I can do the rebase by myself :)

**Pull request merged**